hibernate-community-dialects/src/main/java/org/hibernate/community/dialect/pagination/AltibaseLimitHandler.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+
+package org.hibernate.community.dialect.pagination;
+
+import org.hibernate.dialect.pagination.AbstractLimitHandler;
+import org.hibernate.query.spi.Limit;
+
+/**
+ * Limit handler for {@link org.hibernate.community.dialect.AltibaseDialect}.
+ *
+ * @author Geoffrey park
+ */
+public class AltibaseLimitHandler extends AbstractLimitHandler {
+	public static final AltibaseLimitHandler INSTANCE = new AltibaseLimitHandler();
+
+	AltibaseLimitHandler() {
+	}
+
+	@Override
+	public boolean supportsLimit() {
+		return true;
+	}
+
+	@Override
+	public String processSql(String sql, Limit limit) {
+		if ( supportsLimit() && hasMaxRows( limit ) ) {
+			final boolean hasFirstRow = hasFirstRow( limit );
+			if (hasFirstRow) {
+				int firstRow = limit.getFirstRow();
+				limit.setFirstRow(firstRow + 1);   // In Altibase offset starts from 1
```

Comment:
This is not safe. You should rather extend from `LimitLimitHandler` and override the `limitClause` and soon also the `offsetOnlyClause` (see https://github.com/hibernate/hibernate-orm/pull/7520) to and `return sql + ( hasFirstRow ? " limit 1+?,?" : " limit ?" )`

Reply:
After patching #7520 and modifing AltibaseLimitHandler to override LimitLimitHandler, I could pass NativeQueryLimitOffsetTest test. Thanks!